comment/views/followers.py

```diff
@@ -0,0 +1,61 @@
+from django.core.exceptions import ValidationError
+from django.core.validators import EmailValidator
+from django.http import JsonResponse
+
+from django.views.generic.base import View
+
+from comment.models import Follower
+from comment.mixins import CanSubscribeMixin
+from comment.validators import ContentTypeValidator
+from comment.messages import FollowError
+
+
+class BaseToggleFollowView(ContentTypeValidator):
+    response_class = None
+
+    def get_response_class(self):
+        assert self.response_class is not None, (
+                "'%s' should either include a `response_class` attribute, "
+                "or override the `get_response_class()` method."
+                % self.__class__.__name__
+        )
+        return self.response_class
+
+    def post(self, request, *args, **kwargs):
+        """ Allow authenticated users only, anonymous may be added in the future """
+        email = request.POST.get('email', None)
+        response_class = self.get_response_class()
+        if email:
+            email_validator = EmailValidator(FollowError.EMAIL_INVALID)
+            try:
+                email_validator(email)
+            except ValidationError as error:
+                return response_class({'invalid_email': error.messages}, status=400)
```

Comment:
this can probably be moved to a some sort of `EmailValidator` mixin. At present, this function seems to contain too much of code to easily understand what is being tried to achieve.

Reply:
Why would we reinvent the wheel?

Reply:
I was suggesting about separating this login to a separate function(refactoring).